frontend/test/parsing/testParseSelect.cpp

```diff
@@ -180,6 +180,29 @@ static void test3(Parser* parser) {
   assert(w0->body()->numStmts() == 1);
 }
 
+static void test4(Parser* parser) {
+  ErrorGuard guard(parser->context());
+  auto parseResult = parseStringAndReportErrors(parser, "test1.chpl",
+      "/* c1 */\n"
+      "select foo {\n"
+      "  when x do f1();\n"
+      "  otherwise do f2();\n"
+      "  when y { f3(); }\n"
+      "}\n"
+      "/* c2 */\n");
+  assert(guard.numErrors() == 1);
+  auto mod = parseResult.singleModule();
+  assert(mod);
+  assert(mod->stmt(0)->isComment());
+  assert(mod->stmt(1)->isSelect());
+  assert(mod->stmt(2)->isComment());
+  auto& error = guard.error(0);
+  const char* expected = "'otherwise' clause must follow all 'when' clauses";
```

Comment:
Sorry, I missed this earlier. It's best not to pattern match on the wording on the error message. You should be able to check `error->type() == ErrorType::WhenAfterOtherwise`, which will remain true even if the wording of the error message changes.